youtube_dl/extractor/thisvid.py

```diff
@@ -0,0 +1,102 @@
+# coding: utf-8
+from __future__ import unicode_literals
+import re
+
+from .common import InfoExtractor
+
+
+class ThisVidIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?thisvid\.com/(?P<type>videos|embed)/(?P<id>[A-Za-z0-9-]+/?)'
+    _TESTS = [{
+        'url': 'https://thisvid.com/videos/big-man-walking26/',
+        'md5': '74a11cac6a738f28c1a37945145a3f96',
+        'info_dict': {
+            'id': '2123318',
+            'ext': 'mp4',
+            'title': 'Big Man Walking26',
+            'thumbnail': 'https://media.thisvid.com/contents/videos_screenshots/2123000/2123318/preview.mp4.jpg',
+            'age_limit': 18,
+        }
+    }, {
+        'url': 'https://thisvid.com/embed/2123318/',
+        'md5': '74a11cac6a738f28c1a37945145a3f96',
+        'info_dict': {
+            'id': '2123318',
+            'ext': 'mp4',
+            'title': 'Big Man Walking26',
+            'thumbnail': 'https://media.thisvid.com/contents/videos_screenshots/2123000/2123318/preview.mp4.jpg',
+            'age_limit': 18,
+        }
+    }]
+
+    def _real_extract(self, url):
+        main_id = self._match_id(url)
+        webpage = self._download_webpage(url, main_id)
+
+        # URL decryptor was reversed from version 4.0.4, later verified working with 5.0.1
+        kvs_version = self._html_search_regex(r'<script [^>]+?src="https://thisvid\.com/player/kt_player\.js\?v=(\d+(\.\d+)+)">', webpage, 'kvs_version', fatal=False)
+        if not kvs_version.startswith("5."):
+            self.report_warning("Major version change (" + kvs_version + ") in player engine--Download may fail.")
+
+        title = self._html_search_regex(r'<title\b[^>]*?>(?:Video:\s+)?(.+?)(?:\s+-\s+ThisVid(?:\.com| tube))?</title>', webpage, 'title')
+        # video_id, video_url and license_code from the 'flashvars' JSON object:
+        video_id = self._html_search_regex(r"video_id:\s+'([0-9]+)',", webpage, 'video_id')
+        video_url = self._html_search_regex(r"video_url:\s+'(function/0/.+?)',", webpage, 'video_url')
```

Comment:
Later, you're going to rely on `function/0/...'.split('/')` having at least length 8 (2 + 5 + 1). Make sure it has:
```suggestion
        video_url = self._html_search_regex(r"video_url:\s+'(function/0/(?:[^/']+/){5,}.+?)',", webpage, 'video_url')
```
Or, as `function/0` is never used (I think?),
```suggestion
        video_url = self._html_search_regex(r"video_url:\s+'function/0/((?:[^/']+/){5,}.+?)',", webpage, 'video_url')
```
and later (l.66 currently)
```diff
-    urlparts = video_url.split('/')[2:]
+    urlparts = video_url.split('/')
```